java/ql/src/Security/CWE/CWE-022/TaintedPath.ql

```diff
@@ -17,6 +17,17 @@ import semmle.code.java.dataflow.FlowSources
 import PathsCommon
 import DataFlow::PathGraph
 
+class ContainsDotDotSanitizer extends DataFlow::BarrierGuard {
```

Comment:
Why not extend `MethodAccess`?

Reply:
I did initially, but then you get a couple of methods like `getEnclosingCallable()` that are ambiguously inherited, which means that we'll have to add additional overrides to the class.
Now if only QL allowed us to declare membership of a class without affecting dispatch...